artifactory/utils/pip/dependencies/dependencies.go

```diff
@@ -32,6 +36,73 @@ type Extractor interface {
 	Extract() error
 }
 
+func CreateCompatibleExtractor(pythonExecutablePath string, pipArgs []string) (Extractor, error) {
+	// Check if using requirements file.
+	filePath, err := getRequirementsFilePath(pipArgs)
+	if err != nil {
+		return nil, err
+	}
+	if filePath != "" {
+		return NewRequirementsExtractor(filePath, pythonExecutablePath), nil
+	}
+
+	// Setup.py should be in current dir.
+	filePath, err = getSetuppyFilePath()
+	if err != nil {
+		return nil, err
+	}
+	if filePath != "" {
+		return NewSetupExtractor(filePath, pythonExecutablePath), nil
+	}
+
+	// Couldn't resolve requirements file or setup.py.
+	return nil, errorutils.CheckError(errors.New("Could not find installation file for pip command, the command must contain either '--requirement <file>' or run from the directory containing 'setup.py' file."))
```

Comment:
...command must contain either  '--requirement <file>' or run from from the directory containing 'setup.py' file. --> ...command must include  '--requirement <file>' or be executed from within the directory containing the 'setup.py' file.